unified-runtime/source/adapters/native_cpu/kernel.hpp

```diff
@@ -186,10 +192,26 @@ struct ur_kernel_handle_t_ : RefCounted {
 
   void addPtrArg(void *Ptr, size_t Index) { Args.addPtrArg(Index, Ptr); }
 
+  void addArgReference(ur_mem_handle_t Arg) {
+    Arg->incrementReferenceCount();
+    ReferencedArgs.push_back(Arg);
+  }
+
+private:
+  void removeArgReferences() {
+    for (auto arg : ReferencedArgs)
+      decrementOrDelete(arg);
+  }
+  void takeArgReferences(const ur_kernel_handle_t_ &other) {
```

Comment:
```suggestion
  void copyArgReferences(const ur_kernel_handle_t_ &other) {
```
The name "take<...>" usually means that after the operation completes, what you've taken things from no longer has them, in my experience.